### PR TITLE
[IMM32] Rewrite ImmNotifyIME

### PR DESCRIPTION
**Pull request by katahiromz**
## Purpose

Implementing Japanese input...
JIRA issue: [CORE-11700](https://jira.reactos.org/browse/CORE-11700)

## Proposed changes

- Rewrite `imm32!ImmNotifyIME` function.

### CODE REVIEW

File: dll/win32/imm32/imm.c [hunk 1]
```diff
@@ -2609,23 +2609,29 @@ BOOL WINAPI ImmIsUIMessageW(
 BOOL WINAPI ImmNotifyIME(
   HIMC hIMC, DWORD dwAction, DWORD dwIndex, DWORD dwValue)
 {
-    InputContextData *data = get_imc_data(hIMC);
+    DWORD idImeThread, idThread;
```
Comment (HBelusca) [inline dll/win32/imm32/imm.c:2612]: usual win32 code practice uses `dwThreadId` (or just `dwThread` sometimes) . For example: `dwImeThreadId` etc.

File: dll/win32/imm32/imm.c [hunk 2]
```diff
@@ -2609,23 +2609,29 @@ BOOL WINAPI ImmIsUIMessageW(
...
-    TRACE("(%p, %d, %d, %d)\n",
-        hIMC, dwAction, dwIndex, dwValue);
+    TRACE("ImmNotifyIME(%p, %d, %d, %d)\n", hIMC, dwAction, dwIndex, dwValue);
 
-    if (hIMC == NULL)
+    if (hIMC)
     {
-        SetLastError(ERROR_SUCCESS);
-        return FALSE;
+        idImeThread = Imm32QueryWindow(hIMC, 1);
```
Comment (HBelusca) [inline dll/win32/imm32/imm.c:2621]: "1": magic value? (see this list: https://git.reactos.org/?p=reactos.git;a=blob;f=win32ss/include/ntuser.h;hb=9a42267b06861a057163729ca188438d157f13b8#l2841 )

Comment (katahiromz) [reply dll/win32/imm32/imm.c]: Used `QUERY_WINDOW_...`. OK.

File: dll/win32/imm32/imm.c (hunk 1 above)
Comment (katahiromz) [reply dll/win32/imm32/imm.c]: Renamed. OK.

**Pull request merged**